### ISSUE TITLE
[QST] Not generating labels during training after mapping to empty list

### ISSUE THREAD

**Issue by asos-edharris** [NVIDIA-Merlin/Transformers4Rec#356]
# ❓ Questions & Help

## Details

Following along the example notebooks (TF), I map the training labels to an empty list, as done in the tutorial:
`dataloader.map(lambda X, y: (X, []))`
However, when I then print out the labels during training they remain empty:
`Batch output (y_true) =`
Although, the tutorial says: `The reason we set the targets to [] in the data-loader because the true item labels are computed internally by the MaskSequence class.`
Is there somewhere particular that might be causing this issue that you could suggest?

Thanks!

**Comment by sararb** [comment]
Thank you for your question @asos-edharris!  Could you please share the code used to print out the labels during training? 

To give more context to the tutorial statement:   The labels are computed from the input sequence of item-ids and stored in the MaskSequence class as `masked_targets`.  Then, they are retrieved in NextItemPredictionTask when computing the loss ([here](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/7eac1d0ec2a340866e32cae0119a0bd182e653fd/transformers4rec/tf/model/prediction_task.py#L237) ) or the metrics ([here](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/7eac1d0ec2a340866e32cae0119a0bd182e653fd/transformers4rec/tf/model/prediction_task.py#L269))
